holidays/countries/gibraltar.py

```diff
@@ -0,0 +1,220 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars.gregorian import JAN, MAY, JUN, AUG, SEP
+from holidays.groups import ChristianHolidays, InternationalHolidays, StaticHolidays
+from holidays.observed_holiday_base import (
+    ObservedHolidayBase,
+    SAT_SUN_TO_NEXT_MON,
+    SAT_SUN_TO_NEXT_MON_TUE,
+)
+
+
+class Gibraltar(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, StaticHolidays):
+    """Gibraltar holidays.
+
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Gibraltar>
+        * [Public Holidays 2003](https://web.archive.org/web/20250508170633/https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-7023/version/12-12-2002)
```

Comment:
Due to Gibraltar government website's implementation of its PDF viewer renders its web archive version unusable - for cases like this, we can manually download and reupload the files to the Internet Archive.
```suggestion
        * [Public Holidays 2003](https://archive.org/details/2002s-094)
```

If you don't mind waiting, I can help uploading all of these later + [2000](https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-7021) and [2001](https://www.gibraltarlaws.gov.gi/legislations/bank-and-public-holidays-order-7022) sources (Can't find 2002 ones sadly)